torchao/quantization/subclass.py

```diff
@@ -191,19 +191,46 @@ def __torch_dispatch__(cls, func, types, args, kwargs):
             )
 
 
+@torch._dynamo.allow_in_graph
+def from_qtensor_components_int8dyn(
+    *args,
+    **kwargs
+):
+    return Int8DynamicallyQuantizedLinearWeight(
+        *args,
+        **kwargs,
+    )
+
+class ConstructTensorSubclassInt8Dyn(torch.nn.Module):
+    def __init__(self, tensor_subclass_ctr, *args, **kwargs):
+        super().__init__()
+        self.tensor_subclass_ctr = tensor_subclass_ctr
+        self.args = args
+        self.kwargs = kwargs
+
+    def forward(self, int_data, q_scales):
+        return from_qtensor_components_int8dyn(int_data, q_scales, *self.args, **self.kwargs)
+
+    def right_inverse(self, tensor_subclass_instance):
+        return tensor_subclass_instance.int_data, tensor_subclass_instance.q_scales
```

Comment:
Can you use tensor_flatten for this?

Reply:
this works, thanks. I'll create a parent class to host __init__ and right_inverse